src/protocols/MeterExec.cpp

```diff
@@ -38,44 +43,159 @@ MeterExec::MeterExec(std::list<Option> options)
+		while (i <= config_len && j <= scanf_len) {
+			switch (config_format[i]) {
+					case '$':
+						if (i+1 < config_len) { // introducing a token
+							switch (config_format[i+1]) {
+									case 'v': j += sprintf(scanf_format+j, "%%1$lf"); break;
+									case 'i': j += sprintf(scanf_format+j, "%%2$ms"); break;
+									case 't': j += sprintf(scanf_format+j, "%%3$lf"); break;
+									// Regex is not working with gcc-4.6
+									// (?<identifier>[\w]+): (?<value>[\d]+[,|.]?[?:\d]+) (?<timestamp>\d{10})
+									//case 'v': j += sprintf(scanf_format+j, "([\\d]+[.|,]?[?:\\d]+)"); break;
+									//case 'i': j += sprintf(scanf_format+j, "([\\w]+)"); break;
+									//case 't': j += sprintf(scanf_format+j, "(\\d{10})"); break;
+							}
+							i++;
+						}
+						break;
+
+					case '%':
+						scanf_format[j++] = '%'; // add double %% to escape a conversion identifier
+						// nobreak;
+					default:
+						scanf_format[j++] = config_format[i]; // just copying
+			}
+
+			i++;
+		}
+
+		print(log_debug, "MeterExec::MeterExec: Parsed format string \"%s\" => \"%s\"", name().c_str(), config_format, scanf_format);
+		_format = scanf_format;
 	} catch (vz::OptionNotFoundException &e) {
-		_format = NULL; /* use default format */
+		_format = ""; // use default format
 	} catch (vz::VZException &e) {
-		print(log_error, "Failed to parse format", "");
+		print(log_error, "MeterExec::MeterExec: Failed to parse format", name().c_str());
 		throw;
 	}
 }
 
 MeterExec::~MeterExec() {
+}
 
-	free((void*)_command);
+int MeterExec::open() {
+	print(log_debug, "MeterExec::open: Testing command line '%s': %s", name().c_str(), command(), strerror(errno));
+	_pipe = popen(command(), "r");
 
-	if (_format != NULL) {
-		free((void*)_format);
+	if (_pipe == NULL) {
+		print(log_error, "MeterExec::open: popen(%s): %s", name().c_str(), command(), strerror(errno));
+		return ERR;
 	}
-}
 
-int MeterExec::open() {
+/* Pipe is closed after script complete
+ 	if (_pipe != NULL) {
+		pclose(_pipe);
+	}
+*/
 
-	// TODO implement
-	return ERR;
+	return SUCCESS;
 }
 
 int MeterExec::close() {
-
-	// TODO implement
-	return ERR;
+	return SUCCESS;
 }
 
 ssize_t MeterExec::read(std::vector<Reading> &rds, size_t n) {
-	//meter_handle_exec_t *handle = &mtr->handle.exec;
+	char buffer[256];
+	char *string=0;
+
+	unsigned int i = 0;
+
+	print(log_debug, "MeterExec::read: Calling '%s'", name().c_str(), command());
+	_pipe = popen(command(), "r");
+
+	if (_pipe) {
+		while(!feof(_pipe)) {
+			while (fgets(buffer, 256, _pipe)) {
+				char *nl;
+				if ((nl = strrchr(buffer, '\n'))) *nl = '\0'; // remove trailing newlines
+				if ((nl = strrchr(buffer, '\r'))) *nl = '\0';
+
+				if (_format != "") {
+					double timestamp=-1.0;
+
+					// at least the value has to be read
+					double value=0.0;
+
+					print(log_debug, "MeterExec::read: Reading line: '%s'", name().c_str(), buffer);
+					int found = sscanf(buffer, format(), &value, &string, &timestamp);
```

Comment:
sscanf doesn't allocate the space for string.
see here from the man page (man 3 scanf):
```
    s      Matches  a  sequence  of  non-white-space characters; the next pointer must be a pointer to character array that is long enough to hold the
              input sequence and the terminating null byte ('\0'), which is added automatically.  The input string stops at white space or at the maximum
              field width, whichever occurs first.
```
You should allocate the buffer for string, add max. char and then (below) free,... the memory.
We seem to have this problem in MeterFile as well!